### PR TITLE
Clean optima finding added. Unit test updated

### PR DESCRIPTION
**Pull request by davbyr**
Optima finding code in a clean branch of develop, ready for merging. Methods:

`TIDEGAUGE.find_high_and_low_waters()`
`stats_util.find_maxima()`

Also fixed a small problem with the previous implementation where minima were being output as negative.

Unit test updated

Old pull request closed.

### CODE REVIEW

File: unit_testing/unit_test.py [hunk 1]
```diff
@@ -1121,6 +1121,43 @@
...
+    f = plt.figure()
+    plt.plot(lowestoft2.dataset.time, lowestoft2.dataset.sea_level)
+    plt.scatter(peaks.dataset.time_highs, peaks.dataset.sea_level_highs, c='g')
+    plt.scatter(peaks.dataset.time_lows, peaks.dataset.sea_level_lows, c='r')
+    plt.legend(['Time Series','Maxima','Minima'])
+    plt.title('Tide Gauge Optima at Lowestoft')
+    f.savefig(dn_fig + 'tidegauge_optima.png')
+
+    if check1 and check2 and check3 and check4:
+        print(str(sec) + chr(subsec) + " OK - Tidegauge peaks found")
```
Comment (jpolton) [inline unit_testing/unit_test.py:1156]: peaks --> extrema

File: unit_testing/unit_test.py [hunk 2]
```diff
@@ -1121,6 +1121,43 @@
...
+    else:
+        print(str(sec) + chr(subsec) + " X - Tidegauge peaks")
```
Comment (jpolton) [inline unit_testing/unit_test.py:1158]: peaks --> extrema

File: unit_testing/unit_test.py [hunk 3]
```diff
@@ -1121,6 +1121,43 @@
...
+    check3 = len(peaks.dataset.time_highs) == 19
+    check4 = len(peaks.dataset.time_lows) == 18
+    
+    # Attempt a plot
+    f = plt.figure()
+    plt.plot(lowestoft2.dataset.time, lowestoft2.dataset.sea_level)
+    plt.scatter(peaks.dataset.time_highs, peaks.dataset.sea_level_highs, c='g')
+    plt.scatter(peaks.dataset.time_lows, peaks.dataset.sea_level_lows, c='r')
+    plt.legend(['Time Series','Maxima','Minima'])
+    plt.title('Tide Gauge Optima at Lowestoft')
```
Comment (jpolton) [inline unit_testing/unit_test.py:1152]: Optima --> extrema

File: unit_testing/unit_test.py [hunk 4]
```diff
@@ -1121,6 +1121,43 @@
...
+    f.savefig(dn_fig + 'tidegauge_optima.png')
```
Comment (jpolton) [inline unit_testing/unit_test.py:1153]: Optima --> extrema

**Review by jpolton — APPROVED** [review]
Good job. Accept.

**Review by anwiseNOCL — APPROVED** [review]
No problems for me running this.

File: unit_testing/unit_test.py (hunk 1 above)
Comment (davbyr) [reply unit_testing/unit_test.py]: @jpolton Dealt with all suggestions

**Pull request merged**